expo/expo
# [sqlite] move next to default and old one as legacy

# Why

for sdk 51 to promote next as default and keep the previous version as legacy

# How

- [sqlite] only change ts/js code that move next to default and old as legacy
- [test-suite] default -> legacy, next -> default
- [docs][tools] update doc for default -> legacy, next -> default, 

for details, please check each commits that would help for review.
 
# Test Plan

- run test-suite on ios and android
- run previous test-suite to make sure `import * SQLite from 'expo-sqlite/next';` still work
- check local rendering doc

# Checklist
